test/test_bson.py

```diff
@@ -1362,6 +1363,26 @@ def __int__(self):
         with self.assertRaisesRegex(InvalidBSON, re.compile(re.escape(_DATETIME_ERROR_SUGGESTION))):
             decode(encode({"a": DatetimeMS(small_ms)}))
 
+    def test_array_of_documents_to_buffer(self):
+        # An array where the size cannot be represented as int32.
+        # We replace the size of the array with \xff\xff\xff\x00 which is -221 as an int32.
+        # b"-            \x14\x00\x00\x00\x04a\x00\xff\xff\xff\x00\x100\x00\x01\x00\x00\x00\x00\x00",
+        doc = dict(a=1)
+        buf = _array_of_documents_to_buffer(encode(dict(a=doc)))
```

Comment:
Could this test use a cursor.nextBatch like document `{"0": {}, "1": {}, ...}` rather than  `{"a": {}, "b": {}, ...}`?

Reply:
Done